test/rest/defaults.test.js

```diff
@@ -35,49 +35,127 @@ define(['ably', 'chai'], function (Ably, chai) {
+        'main.realtime.ably.net',
+      );
+      expect(Defaults.getHost(normalisedOptions, 'main.realtime.ably.net', true)).to.equal('main.realtime.ably.net');
 
       helper.recordPrivateApi('call.Defaults.getPort');
       expect(Defaults.getPort(normalisedOptions)).to.equal(443);
     });
 
     /**
-     * @spec TO3k1
-     * @spec TO3k2
-     * @spec TO3k3
+     * @spec TO3k8
      * @spec TO3k4
      * @spec TO3k5
      * @spec TO3k6
      * @spec TO3d
-     * @spec RSC15h
-     * @specpartial RSC11 - test default value for restHost
-     * @specpartial RTN2 - test default value for realtimeHost
-     * @specpartial RSC15e - primary host for REST is restHost
-     * @specpartial RTN17a - primary host for realtime is realtimeHost
-     * @specpartial RSC11b - test with environment set to 'production'
-     * @specpartial RSC15g2 - test with environment set to 'production'
-     * @specpartial RTC1e - test with environment set to 'production'
+     * @spec REC1a
      */
-    it('Init with production environment', function () {
+    it('Init with given endpoint', function () {
       const helper = this.test.helper;
 
       helper.recordPrivateApi('call.Defaults.normaliseOptions');
-      var normalisedOptions = Defaults.normaliseOptions({ environment: 'production' }, null, null);
+      var normalisedOptions = Defaults.normaliseOptions({ endpoint: 'nonprod:sandbox' }, null, null);
 
-      expect(normalisedOptions.restHost).to.equal('rest.ably.io');
-      expect(normalisedOptions.realtimeHost).to.equal('realtime.ably.io');
       expect(normalisedOptions.port).to.equal(80);
       expect(normalisedOptions.tlsPort).to.equal(443);
-      expect(normalisedOptions.fallbackHosts.sort()).to.deep.equal(Defaults.FALLBACK_HOSTS.sort());
+      expect(normalisedOptions.fallbackHosts.sort()).to.deep.equal(
+        Defaults.getEndpointFallbackHosts('nonprod:sandbox').sort(),
+      );
       expect(normalisedOptions.tls).to.equal(true);
 
       helper.recordPrivateApi('call.Defaults.getHosts');
       expect(Defaults.getHosts(normalisedOptions).length).to.deep.equal(4);
       expect(Defaults.getHosts(normalisedOptions)[0]).to.deep.equal(normalisedOptions.restHost);
       helper.recordPrivateApi('call.Defaults.getHost');
-      expect(Defaults.getHost(normalisedOptions, 'rest.ably.io', false)).to.deep.equal('rest.ably.io');
-      expect(Defaults.getHost(normalisedOptions, 'rest.ably.io', true)).to.deep.equal('realtime.ably.io');
+      expect(Defaults.getHost(normalisedOptions, 'sandbox.realtime.ably-nonprod.net', false)).to.deep.equal(
+        'sandbox.realtime.ably-nonprod.net',
+      );
+      expect(Defaults.getHost(normalisedOptions, 'sandbox.realtime.ably-nonprod.net', true)).to.deep.equal(
+        'sandbox.realtime.ably-nonprod.net',
+      );
+
+      helper.recordPrivateApi('call.Defaults.getPort');
+      expect(Defaults.getPort(normalisedOptions)).to.equal(443);
+    });
+
+    /**
+     * @spec TO3k8
```

Comment:
I notice that we have multiple `@spec` tags for a single spec point. Did you see the guidance about how to use these tags?

https://github.com/ably/ably-js/blob/f6fbe350c6c84f69811c0896e4999f94c2d80ef4/CONTRIBUTING.md?plain=1#L63-L66

Please could you check that you're using these tags correctly? It might be that `@specpartial` is more appropriate in some places, in which case it would be good to accompany them with an explanatory comment.

Reply:
I don't think I am. I copied and pasted these tests just with a different value for the `endpoint` client option. Is there a better way I should structure these tests?